### PR TITLE
Added unit tests for converter package.

### PR DESCRIPTION
**Pull request by svInfra17**
## Description
Added ginkgo unit tests for converter package.

```release-note
None required
```

### CODE REVIEW

File: pkg/converter/converter_suite_test.go [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package converter
```
Comment (caseydavenport) [inline pkg/converter/converter_suite_test.go]: Sorry... I was wrong about this - it should be `package converter_test`

File: Makefile [hunk 2]
```diff
@@ -69,11 +72,11 @@ binary-containerized: vendor
 	    make OS=$(OS) ARCH=$(ARCH) binary'
 
 ###############################################################################
-# Test targets 
+# Test tarets
```
Comment (caseydavenport) [inline Makefile]: typo: tarets

**Comment by caseydavenport** [comment]
Looks like the tests are failing in semaphore because ginkgo isn't installed.

Take a look at what we do in libcalico - https://github.com/projectcalico/libcalico-go/blob/master/Makefile#L31

We should have a test-containerized target which gets called by CI

File: Makefile [hunk 3]
```diff
@@ -73,7 +76,16 @@ binary-containerized: vendor
 ###############################################################################
 
 ## Runs all tests - good for CI. 
-ci: clean docker-image st # TODO: ut
+ci: clean docker-image ut st
```
Comment (caseydavenport) [inline Makefile]: This needs to call the `test-containerized` target instead of `ut`

File: Makefile [hunk 4]
```diff
@@ -73,7 +76,16 @@ binary-containerized: vendor
 ###############################################################################
 
 ## Runs all tests - good for CI. 
-ci: clean docker-image st # TODO: ut
+ci: clean docker-image ut st 
+
+## Run the tests in a container. Useful for CI, Mac dev.
+test-containerized: vendor run-etcd run-k8s-apiserver
+	-mkdir -p .go-pkg-cache
```
Comment (caseydavenport) [inline Makefile]: The indentation looks off here.

File: pkg/converter/namespace_converter_test.go [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+	. "github.com/onsi/gomega"
+	"github.com/projectcalico/k8s-policy/pkg/converter"
+	"github.com/projectcalico/libcalico-go/lib/api"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8sapi "k8s.io/client-go/pkg/api/v1"
+)
+
+var _ = Describe("NamespaceConverter", func() {
+	nsConverter := converter.NewNamespaceConverter()
+	Context("should parse a Namespace to a Profile", func() {
```
Comment (caseydavenport) [inline pkg/converter/namespace_converter_test.go]: I think this is missing a test for `GetKey` (as well as the other conversion files).  Should be a quick one to add.

**Comment by caseydavenport** [comment]
@svInfra17 just a couple more comments.  Once those are addressed and the tests are passing on semaphore I'm happy to merge.

**Comment by caseydavenport** [comment]
Also looks like this needs a rebase.

**Comment by caseydavenport** [comment]
@svInfra17 looks like a test failure:

```
panic: assignment to entry in nil map

goroutine 1 [running]:
github.com/projectcalico/k8s-policy/pkg/converter.(*podConverter).Convert(0x1cefea0, 0x1421ac0, 0xc420331500, 0x8, 0x8, 0x138cf20, 0x101000000000000)
	github.com/projectcalico/k8s-policy/pkg/converter/_obj/pod_converter.go:36 +0x300
github.com/projectcalico/k8s-policy/pkg/converter_test.glob..func2.1()
	/go/src/github.com/projectcalico/k8s-policy/pkg/converter/pod_converter_test.go:27 +0x125
github.com/projectcalico/k8s-policy/vendor/github.com/onsi/ginkgo/internal/suite.(*Suite).PushContainerNode(0xc4202de370, 0x144ef8c, 0x12, 0xc4205d1d98, 0x0, 0x1c3e552, 0x4f, 0x29, 0xc4203d7e00, 0x1d5)
	/go/src/github.com/projectcalico/k8s-policy/vendor/github.com/onsi/ginkgo/internal/suite/suite.go:144 +0x1c2
github.com/projectcalico/k8s-policy/vendor/github.com/onsi/ginkgo.Context(0x144ef8c, 0x12, 0xc4205d1d98, 0xc4202cfd70)
	/go/src/github.com/projectcalico/k8s-policy/vendor/github.com/onsi/ginkgo/ginkgo_dsl.go:311 +0xbf
github.com/projectcalico/k8s-policy/pkg/converter_test.glob..func2()
	/go/src/github.com/projectcalico/k8s-policy/pkg/converter/pod_converter_test.go:41 +0x87
github.com/projectcalico/k8s-policy/vendor/github.com/onsi/ginkgo/internal/suite.(*Suite).PushContainerNode(0xc4202de370, 0x144636a, 0xc, 0x14ae358, 0x0, 0x1c3e552, 0x4f, 0x60, 0xc4200ac780, 0xb6)
	/go/src/github.com/projectcalico/k8s-policy/vendor/github.com/onsi/ginkgo/internal/suite/suite.go:144 +0x1c2
github.com/projectcalico/k8s-policy/vendor/github.com/onsi/ginkgo.Describe(0x144636a, 0xc, 0x14ae358, 0x1131201)
	/go/src/github.com/projectcalico/k8s-policy/vendor/github.com/onsi/ginkgo/ginkgo_dsl.go:282 +0xbf
github.com/projectcalico/k8s-policy/pkg/converter_test.init()
	/go/src/github.com/projectcalico/k8s-policy/pkg/converter/pod_converter_test.go:96 +0xc2
main.init()
	github.com/projectcalico/k8s-policy/pkg/converter/_test/_testmain.go:142 +0x5f
```

**Comment by svInfra17** [comment]
```
panic: assignment to entry in nil map
```
Will get addressed by PR (#123)

**Comment by caseydavenport** [comment]
@svInfra17 I think you meant https://github.com/projectcalico/k8s-policy/pull/127, which is merged now.  Can you please rebase to pick up that commit and re-enable the `ut` target in CI so that we can see the tests pass before merging?

**Comment by svInfra17** [comment]
@caseydavenport , Yes after rebase build no. 12 has passed successfully with 47 specs.

**Pull request merged**